department-of-veterans-affairs/abd-vro
# Pre-commit hook: configure and add gitleaks

## What was the problem?


No automation to prevent secrets from being committed to git.

- #657

## How does this fix it?


Add [gitleaks](https://github.com/zricethezav/gitleaks#pre-commit) as a pre-commit hook to help prevent secrets from getting to our GitHub repo.


## How to test this PR

- [Install pre-commit](https://pre-commit.com/#install)
- Run: `pre-commit install`. This creates a `.git/hooks/pre-commit` file.
- Make and stage a change, try to commit. Expect pre-commit to run, failing the commit if a secret is included or some other fix is needed.

Example failed commit attempt:
```
Check for useless excludes...............................................Passed
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check yaml...............................................................Passed
check for merge conflicts................................................Passed
check xml............................................(no files to check)Skipped
flake8...................................................................Passed
isort....................................................................Failed
- hook id: isort
- files were modified by this hook

Fixing /Users/yoomlam/dev/abd-vro/service-python/assessclaimdc7101v2/src/lib/main.py

Detect hardcoded secrets.................................................Passed
```

Try to add a Slack webhook URL (`https://hooks.slack.com/services/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`):
```
Check for useless excludes...........................(no files to check)Skipped
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check yaml...........................................(no files to check)Skipped
check for merge conflicts................................................Passed
check xml............................................(no files to check)Skipped
flake8...............................................(no files to check)Skipped
isort................................................(no files to check)Skipped
Detect hardcoded secrets.................................................Failed
- hook id: gitleaks
- exit code: 1

○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

Finding:     String s="REDACTED";
Secret:      REDACTED
RuleID:      slack-web-hook
Entropy:     5.063107
File:        shared/lib-camel-connector/src/main/java/gov/va/vro/camel/CamelConfiguration.java
Line:        34
Fingerprint: shared/lib-camel-connector/src/main/java/gov/va/vro/camel/CamelConfiguration.java:slack-web-hook:34

9:30AM INF 1 commits scanned.
9:30AM INF scan completed in 67.7ms
9:30AM WRN leaks found: 1
```
